src/test/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSFilterTest.java

```diff
@@ -0,0 +1,243 @@
+import htsjdk.samtools.SAMSequenceRecord;
+import htsjdk.samtools.util.SequenceUtil;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.io.LineIterator;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.engine.spark.SparkContextFactory;
+import org.broadinstitute.hellbender.utils.read.ArtificialReadUtils;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.testng.Assert;
+import org.testng.TestException;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import scala.Tuple2;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class PSFilterTest extends CommandLineProgramTest {
+
+    private final String BWA_IMAGE_PATH = "src/test/resources/" + PSFilter.class.getPackage().getName().replace(".", "/") + "/hg19mini.fasta.bwa_image";
+
+    final static List<List<String>> readFastq(final File fastqFile) {
+        try {
+            final LineIterator itr = IOUtils.lineIterator(new FileReader(fastqFile));
+            final List<List<String>> seqs = new ArrayList<>(100);
+            while (itr.hasNext()) {
+                final List<String> read = new ArrayList<>(3);
+                String readName = itr.next();
+                if (readName.endsWith("/1") || readName.endsWith("/2")) {
+                    readName = readName.substring(0, readName.length() - 2);
+                }
+                read.add(readName);
+                read.add(itr.next());
+                itr.next();
+                read.add(itr.next());
+                seqs.add(read);
+            }
+            return seqs;
+        } catch (IOException e) {
+            logger.error("Could not open test file " + fastqFile.getAbsolutePath(), e);
+        }
+        throw new TestException("Invalid fastq test file " + fastqFile);
+    }
+
+    final static List<GATKRead> getReadsFromFastq(final File fastqFile) {
+        final List<List<String>> fastqList = readFastq(fastqFile);
+        return fastqList.stream().map(list -> {
+            final String name = list.get(0);
+            final byte[] bases = list.get(1).getBytes();
+            final byte[] qual = list.get(2).getBytes();
+            final GATKRead read = ArtificialReadUtils.createArtificialRead(bases, qual, "*");
+            read.setName(name);
+            return read;
+        }).collect(Collectors.toList());
+    }
+
+    final static List<GATKRead> getPairedReadsFromFastq(final File fastqFile1, final File fastqFile2) {
+        final List<GATKRead> readList1 = getReadsFromFastq(fastqFile1);
+        final List<GATKRead> readList2 = getReadsFromFastq(fastqFile2);
+        readList1.stream().forEach(read -> {
```

Comment:
you can leave out ```stream()```:  collection has a forEach method.  (same comment 4 lines below.)